pkg/ingress/controller.go

```diff
@@ -653,6 +653,10 @@ func (c *Controller) checkClusterHealth(ctx context.Context, cancelFunc context.
 		if err != nil {
 			// Finally failed health check, then give up leader.
 			log.Warnf("failed to check health for default cluster: %s, give up leader", err)
+			c.apiServer.HealthState.Lock()
+			defer c.apiServer.HealthState.Unlock()
+
+			c.apiServer.HealthState.Err = err
```

Comment:
We may put the health state to the `cluster`? So that if we have multiple APISIX clusters (in the future), we don't have to change the existing codes.

Reply:
Any suggestions about it? Inserting the health state struct into `cluster` is kinda unnecessary I think (we had `cacheSyncErr`). 
We might wanna add a function, be like `APISIX.Health()`, it returns the health state of all clusters.
Or we can just leave it in here (`checkClusterHealth`).

Reply:
@tokers  @mayocream I will merge this PR first, here you can continue to discuss in order to reach agreement.